# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [2]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [3]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [5]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [6]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [7]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [8]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [9]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [10]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [11]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [12]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [13]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [14]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [15]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [16]:
data.duplicated().sum()

71

In [17]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [18]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [19]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [20]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [21]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [22]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [23]:
import pandas as pd
data_pivot = data.pivot_table(index='children', values = 'debt', aggfunc = ['count', 'sum', 'mean'])
data_pivot.columns = ['Общее кол-во займов', 'Общая сумма должников', 'Cреднее значение в категории: \"Задолженность по возврату кредитов"']
data_pivot.index.name='Количество детей в семье'
data_pivot['Cреднее значение в категории: \"Задолженность по возврату кредитов"'] = [f'{x:,.2%}' for x in data_pivot['Cреднее значение в категории: \"Задолженность по возврату кредитов"']]
data_pivot = data_pivot.sort_values('Количество детей в семье', ascending = True)
data_pivot.head(10)

Общее кол-во займов  Общая сумма должников  \
Количество детей в семье                                               
0                                       14091                   1063   
1                                        4808                    444   
2                                        2052                    194   
3                                         330                     27   
4                                          41                      4   
5                                           9                      0   

                         Cреднее значение в категории: "Задолженность по возврату кредитов"  
Количество детей в семье                                                                     
0                                                                     7.54%                  
1                                                                     9.23%                  
2                                                                     9.45%                  
3                                                                     8.18%                  
4                                                                     9.76%                  
5                                                                     0.00%

**Вывод:** 



В отношении семей с 0, 1 и 2 детьми прослеживается наглядная зависимость: **чем меньше детей, тем меньше процент просрочки по кредиту.** 


    
Кол-во займов у каждой категории с увеличением кол-ва детей также заметно уменьшается.

    
    
Рассмотрим градацию по пункту "Проценту просрочки кредита":

1) 2 ребенка в семье - общее кол-во займов: 2052; процент просрочки: 9.45%


2) 1 ребенок в семье - общее кол-во займов: 4808; процент просрочки: 9.23%


3) О детей в семье - общее кол-во займов: 14091; процент просрочки: 7.54%
    
    
Семьям с 3 детьми, возможно, можеть быть легче выплачивать кредит, в связи с оказываемой поддержкой государства. Т.к. семья с 3

    
детьми имеет статус многодетной, различные субсидии и выделенные средства могут благоприятно сказываться на выплате кредита в срок.
    
По семьям с 4 и 5 детьми можно сделать два вывода:
    
a) Исходя из имеющихся данных, семья с 4 детьми подвергается большой финансовой нагрузке и в результате этого, не справляется с выплатой кредита в срок. 
    
Клиенты, у которых 5 детей, берут кредиты реже, но выполняют обязательства без просрочек, т.к.подходят к вопросу кредитования взвешенно и ответственно.
    
    
b) Для более подробного и объективного анализа необходимо больше данных. Тогда, возможно, итоговые цифры могут поменяться.
    



#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [24]:
import pandas as pd
data_pivot_family = data.pivot_table(index='family_status', values = 'debt', aggfunc = ['count', 'sum', 'mean'])
data_pivot_family.columns = ['Общее кол-во займов', 'Общая сумма должников', 'Cреднее значение в категории: \"Задолженность по возврату кредитов"']
data_pivot_family.index.name='Семейное положение'
data_pivot_family['Cреднее значение в категории: \"Задолженность по возврату кредитов"'] = [f'{x:,.2%}' for x in data_pivot_family['Cреднее значение в категории: \"Задолженность по возврату кредитов"']]
data_pivot_family = data_pivot_family.sort_values('Cреднее значение в категории: \"Задолженность по возврату кредитов"', ascending = False)
data_pivot_family.head(10)

Общее кол-во займов  Общая сумма должников  \
Семейное положение                                                  
Не женат / не замужем                 2796                    273   
гражданский брак                      4134                    385   
женат / замужем                      12261                    927   
в разводе                             1189                     84   
вдовец / вдова                         951                     63   

                      Cреднее значение в категории: "Задолженность по возврату кредитов"  
Семейное положение                                                                        
Не женат / не замужем                                              9.76%                  
гражданский брак                                                   9.31%                  
женат / замужем                                                    7.56%                  
в разводе                                                          7.06%                  
вдовец / вдова                                                     6.62%

**Вывод:** 
В данном случае, **четкая зависимость между семейным положением и возвратом кредита в срок не прослеживается**

— На основании данных, можно сделать вывод, что люди, которые состоят в отношениях (женат/замужем) могут иметь большую потребность в кредите, однако процент просрочек может быть как средний, так и высокий (гражданский брак). 
Об этом говорит среднее значение задолженности и общее кол-во займов. Предполагаю, что связано может быть с тем, что финансовые обязательства можно разделить с партнером.

— Категориям без партнера может быть сложнее справляться с разными жизненными обстоятельствами: невыгодные условия кредита, некорректная идентификация собственной покупательской способности, болезнь, потеря работы. В случае, если свои финансовые обязательства не на кого переложить, возможна просрочка кредита.

— Примечательно, что у категорий «вдова\вдовец» и «в разводе» минимальные показатели займов и просрочек. Это может говорит о том, что такие клиенты относятся к кредитам более осторожно и ответственно

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [25]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass
data['total_income_category'] = data['total_income'].apply(categorize_income)
data_pivot_income = data['total_income_category']
data_pivot_income = data.pivot_table(index='total_income_category', values = 'debt', aggfunc = ['count', 'sum', 'mean'])
data_pivot_income.columns = ['Общее кол-во займов', 'Общая сумма должников', 'Cреднее значение в категории: \"Задолженность по возврату кредитов"']
data_pivot_income.index.name='Ежемесячный доход в руб.'
data_pivot_income['Cреднее значение в категории: \"Задолженность по возврату кредитов"'] = [f'{x:,.2%}' for x in data_pivot_income['Cреднее значение в категории: \"Задолженность по возврату кредитов"']]
data_pivot_income = data_pivot_income.sort_values('Общее кол-во займов', ascending =False)
data_pivot_income.head(10)

Общее кол-во займов  Общая сумма должников  \
Ежемесячный доход в руб.                                               
C                                       15921                   1353   
B                                        5014                    354   
D                                         349                     21   
A                                          25                      2   
E                                          22                      2   

                         Cреднее значение в категории: "Задолженность по возврату кредитов"  
Ежемесячный доход в руб.                                                                     
C                                                                     8.50%                  
B                                                                     7.06%                  
D                                                                     6.02%                  
A                                                                     8.00%                  
E                                                                     9.09%

**Вывод:** 
Условные обозначения ежемесячного дохода:

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.

**Зависимость между уровнем дохода и просрочкой прослеживается слабо. Нельзя утверждать, что тенденция «чем больше доход - тем меньше просрочек по кредиту» или наоборот, подтверждается в указанном примере.**

По данным, можно сделать вывод, что гражданам с категорией заработка С и E справляться с возвратом кредита в срок сложнее всего. Вместе с этим, заметно что клиенты, с заработком категории. А также имеют высокий процент просрочки.  Это может говорить о том, что:

**1)Для репрезентативной выборки нужно больше данных.** Нельзя делать вывод о категориях граждан Е и А, т.к. в таблице с данными их представляют 22 и 25 человек. Процент может быть высоким из-за изначально маленького кол-ва в категориях.


**2)Нам не до конца известны условия кредитования и источники дохода граждан.**


**3)Граждане с категорией заработка D справляются с возвратом кредита в срок лучше всего.** Предполагаю, что данная группа кредитуемых меньше всего склонна к импульсным(эмоциональным) покупкам, которые могут подразумевать кредититование.







Рассмотрим категории с наибольшими показателями по общему кол-ву займов и проценту просрочек.

    
Здесь мы видим, что граждане категориями заработка "B" и "C" обращается к услугам банка чаще всего. Имея при этом высокий процент просрочек кредита. 

Граждане с категорией заработка D справляются с возвратом кредита в срок лучше всего. Но и кол-во займов у категории "D" заметно меньше, чем у остальных.
    
    
Из этого можно сделать вывод, что зависимость между заработком и возвратом кредита в срок есть. Люди имеющие средний заработок чаще берут кредиты и чаще их просрачивают.
 
    
Люди, с достатком ниже среднего, возможно, имеют другой взгляд на кредитования как инструмент, поэтому реже обращаются в банк. И если взяли кредит, стараются в срок его выплатить.
    
    
По категориям A и E можно также сделать два вывода:

a) Исходя из имеющихся данных, можно предположить, что эти категории не являются целевой аудитории банка. Категории A просто 
    
нет необходимости брать кредит на что-то. Их достаток позволяет им не пользоваться услугами банка. Категория "E" наоборот, возможно осознает сумму переплаты в результате кредитования и предпочитают не переплачивать, а совершить покупку позже, накопив средства
    
b) Данные категориям A и E не являются репрезентативными. Для полноценного анализа этих категорий нужно больше данных.


#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [27]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'
    
data_pivot_purpose = data['purpose_category']
data_pivot_purpose = data.pivot_table(index='purpose_category', values = 'debt', aggfunc = ['count', 'sum', 'mean'])
data_pivot_purpose.columns = ['Общее кол-во займов', 'Общая сумма должников', 'Cреднее значение в категории: "Задолженность по возврату кредитов"']
data_pivot_purpose.index.name='Цель получения кредита'
data_pivot_purpose['Cреднее значение в категории: "Задолженность по возврату кредитов"'] = [f'{x:,.2%}' for x in data_pivot_purpose['Cреднее значение в категории: "Задолженность по возврату кредитов"']]
data_pivot_purpose = data_pivot_purpose.sort_values('Cреднее значение в категории: "Задолженность по возврату кредитов"', ascending = False)
data_pivot_purpose.head(20)

Общее кол-во займов  Общая сумма должников  \
Цель получения кредита                                                 
операции с автомобилем                   4279                    400   
получение образования                    3988                    369   
проведение свадьбы                       2313                    183   
операции с недвижимостью                10751                    780   

                         Cреднее значение в категории: "Задолженность по возврату кредитов"  
Цель получения кредита                                                                       
операции с автомобилем                                                9.35%                  
получение образования                                                 9.25%                  
проведение свадьбы                                                    7.91%                  
операции с недвижимостью                                              7.26%

**Вывод:** 

**Здесь мы видим, что цели на кредит имеют прямое влияние на возврат кредита в срок.**

1)**Самый востребованный кредит (кредит на операции с недвижимостью)** имеет самый меньший процент просрочек. Это может быть связано с тем, что к такому виду кредита люди относятся очень серьезно, т.к. это длительный и масштабный кредит, накладывающий серьезные финансовые обязательства. Выплаты на недвижимость люди часто ставят в приоритет.

2)**Кредит на проведение свадьбы** также имеет небольшой процент просрочек. Это может быть связано с тем, что полученные со свадьбы средства, чаще всего, покрывают взятый кредит, что способствует возврату денег в установленный срок.

3)**Второй по популярности кредит (кредит на операции с автомобилем)** имеет самый высокий процент просрочек. Это может быть связано с тем, что покупка автомобиля является более импульсной покупкой, чем приобретения жилья. Также, просрочке могут способствовать различные факторы, связанные с авто: аварии, ремонт, обслуживание и тд. Это может привести к непредвиденным расходам и помешать выплатить кредит своевременно.

4)**Еще одной целью кредитования с большим кол-вом просрочек является кредит на получение опобразования**. Предполагаю, что основная целевая аудитория таких кредитов – студенты или те, кто собирается пойти учиться. Совмещение работы и полноценной учебы может сказаться на выплате кредита в срок. 

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

**— Человеческий фактор**(данные не ввели случайно или намеренно, либо заполнены с ошибкой. Это может привести к дальнейшему появлению пропусков в других ячейках данных)


**— Пропуски могут появится, если в остальных ячейках есть исчерпывающая информация**


**— Изначальная ошибка в коде, с помощью которого выгружали исходные данные.** Она также может привести к неточностям в выгрузке исходных данных


#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

Так как **медиана - это число в выборке, которая меньше одной половины элементов и больше другой, она корректно характеризует данные, в случае, когда некоторые значения сильно выделяются среди большинства.** 

**Лучше работает в случае заполнения пропусков в количественных переменных** так как **учитывает разброс данных**, в отличие от средней арифметической.

### Шаг 4: общий вывод.

Было проведено исследование на тему надежности заёмщиков на основании статистики о платежеспособности клиентов банка.

Что сделано: 

**1)Изучена общая информации в данных**

**2)Обработаны пропуски в данных**

**3)Удалены аномальные значения**

**4)Изменены типы данных**

**5)Проработаны дубликаты**

**6)Проведена категоризации данных**

На основании данного исследования будет строиться модель кредитного скоринга. Для этого более подробно были изучены следующие факторы, которые могут влиять на платежеспособность клиентов и также зависимости в данных категориях:
Кол-во детей
Семейное положение
Уровень дохода
Цель кредитования


Рекомендации:


**1)Формировать более подробный портрет предполагаемого заемщика.**
Для эффективной работы банка, скорингу необходимо сразу отличить «плохого» клиента, которому заявка сразу отклоняется от «хорошего», кому заявка согласовывается.

Расширить критерии для анализа клиента. Например, сделать более подробный анализ предоставления кредита другими банками ранее. Сделать более точную детализацию. На данном этапе, в графе `debt` нет информации о причинах просрочек кредита. В данных, на основе которых проводился анализ, не дают в полной мере точно зафиксировать какую-либо зависимость в вышеуказанных категориях.

Что нам известно о результатах ислследования зависимоcти возврата кредита в срок и различных факторов:


**1)Кол-во детей, в большинстве случаев, влияет на возврат кредита в срок. Как показало исследование, чем меньше детей в семье, тем меньше процент просрочки по кредиту.**


**2)Семейное положение также влияет на возврат кредита в срок. В результате исследования данных, мы выяснили, что люди состоящие или состоявшие в браке ответственнее относятся к кредитованию.**


**3)Ежемесячынй доход также влияет на возврат кредита в срок. Основная целевая аудитория - средний класс.**


**4)Цели на кредит имеют прямое влияние на возврат кредита в срок. В зависимости от цели кредита, кредитуемый, распредиляет приоритеты.**


**Учитывая эти данные, попробуем воспроизвести примерный портрет "хорошых клиентов" и "плохих клиентов":**

Кейс с "хорошим клиентом": семья с 1 ребенком и средним достатком берет кредит на квартиру. Учитывая факторы выше, они, с высокой вероятностью, вернуть кредит в срок.


Кейс с "плохим клиентом": семья с 2 детьми и средним достатком берет кредит на машину. Вполне вероятно, что с возвратом кредитра в срок могут возникнуть проблемы.


Кейс с "хорошим клиентом": вдова\человек в разводе с низким достатком берет кредит на Iphone или телевизор. Учитывая факторы выше, она(он), скорее всего, вернуть кредит в срок


Кейс с "плохим клиентом": не женатый клиент имеет средний достаткок и берет кредит на машину. Учитывая выводы выше, возможно, что с возвратом кредитра в срок могут возникнуть проблемы


**2)Рассчитывать индивидуальные условия кредита (процент, лимит, срок, график погашения).** По исходным данным, можно предположить, что банк предлагает всем одинаковые условия кредитования и пытается найти закономерность в просрочках. В данном случае, это, предположительно, борьба со следствием, а не с причиной.
